src/chttpd_handlers.erl

```diff
@@ -51,6 +54,20 @@ design_handler(HandlerKey, DefaultFun) ->
 %% Internal Function Definitions
 %% ------------------------------------------------------------------
 
+is_disabled(KeyBin) ->
+    BlacklistConfig = config:get("chttpd", "url_handler_blacklist", []),
```

Comment:
@janl It's not much different from yours:
```
[chttpd_design_handler_blacklist]
foo = not_found

[chttpd_global_handler_blacklist]
bar = not_found

[chttpd_db_handler_blacklist]
baz = not_implemented

```

except that here you can direct match by url key as it is an option name. In your case you'll have to parse and scan list everytime, for hits and misses. 

And reason to have any cache goes away since config structure becomes more friendly.